fedbiomed/researcher/cli.py

```diff
@@ -86,36 +92,31 @@ def initialize(self):
         """Initializes Researcher CLI"""
 
 
-        class ConfigNameActionResearcher(ConfigNameAction):
+        class ComponentDirectoryActionResearcher(ComponentDirectoryAction):
             _this = self
             _component = ComponentType.RESEARCHER
 
-            def set_component(self, config_name: str) -> None:
+            def set_component(self, component_dir: str) -> None:
                 """Import configuration
 
                 Args:
                     config_name: Name of the config file for the component
                 """
-                config_file = os.environ.get("CONFIG_FILE")
-                if config_file:
-                    os.environ["FBM_RESEARCHER_CONFIG_FILE"] = config_file
-
+                os.environ["FBM_RESEARCHER_COMPONENT_ROOT"] = os.path.abspath(
+                        component_dir
+                )
                 module = importlib.import_module("fedbiomed.researcher.config")
                 self._this.config = module.config
 
+        super().initialize()
 
-        # Config parameter is not necessary. Python client (user in jupyter notebook)
-        # will always use default config file which is `researcher_config`
-        # However, this argument will play important role once researcher back-end (orhestrator)
-        # and researcher is seperated
         self._parser.add_argument(
-            "--config",
-            "-cf",
+            "--path",
+            "-p",
             nargs="?",
-            action=ConfigNameActionResearcher,
-            default=DEFAULT_CONFIG_FILE_NAME_RESEARCHER,
-            help="Name of the config file that the CLI will be activated for. "
-                 f"Default is '{DEFAULT_CONFIG_FILE_NAME_RESEARCHER}'."
+            action=ComponentDirectoryActionResearcher,
+            default="fbm-researcher",
+            help="Name of the config file that the CLI will be activated for. Default "
+                "is 'config_researcher.ini'."
```

Comment:
is there a way to specify the jupyter notebook ip address through the config file? or through the command line?

Reply:
and its port?

Reply:
Hi @ybouilla,
As I remember there is no option for that.

Reply:
oh ok i thought there was an option for that in the previous environ class of fbm, but i may not recall correctly